x-pack/plugins/index_lifecycle_management/public/application/sections/edit_policy/components/phases/shared_fields/data_tier_allocation_field/components/data_tier_allocation.tsx

```diff
@@ -161,7 +161,20 @@ const getSelectOptions = (phase: PhaseWithAllocation, disableDataTierOption: boo
   ].filter(Boolean) as SelectOptions[];
 
 export const DataTierAllocation: FunctionComponent<SharedProps> = (props) => {
-  const { phase, hasNodeAttributes, disableDataTierOption, isLoading } = props;
+  const {
+    phase,
+    hasNodeAttributes,
+    isCloudEnabled,
+    isUsingDeprecatedDataRoleConfig,
+    isLoading,
+  } = props;
+
+  /**
+   * On Cloud we want to disable the data tier allocation option when we detect that we are not
+   * using node roles in our Node config yet. See {@link ListNodesRouteResponse} for information about how this is
+   * detected.
+   */
+  const disableDataTierOption = Boolean(isCloudEnabled && isUsingDeprecatedDataRoleConfig);
```

Comment:
This was originally being defined in `data_tier_allocation_field.tsx` and passed in as a prop. I refactored the code to define this value locally, derived from the underlying conditions, to make it easier to reason about when reading this file. There's no change to the original behavior.